src/Data/Term.hs

```diff
@@ -171,3 +151,35 @@ instance (ToJSON b, ToJSONFields a, ToJSONFields1 f) => ToJSONFields (TermF f a
 instance (ToJSON b, ToJSONFields a, ToJSONFields1 f) => ToJSON (TermF f a b) where
   toJSON = object . toJSONFields
   toEncoding = pairs . mconcat . toJSONFields
+
+
+class IsTerm term where
+  type Syntax term :: * -> *
+
+  toTermF :: term ann -> TermF (Syntax term) ann (term ann)
+  fromTermF :: TermF (Syntax term) ann (term ann) -> term ann
+
+
+termAnnotation :: IsTerm term => term ann -> ann
+termAnnotation = termFAnnotation . toTermF
+
+termOut :: IsTerm term => term ann -> Syntax term (term ann)
+termOut = termFOut . toTermF
+
+projectTerm :: (f :< syntax, Sum syntax ~ Syntax term, IsTerm term) => term ann -> Maybe (f (term ann))
+projectTerm = Sum.project . termOut
+
+
+-- | Build a term from its annotation and syntax.
+termIn :: IsTerm term => ann -> Syntax term (term ann) -> term ann
+termIn = fmap fromTermF . In
```

Comment:
Thought: would a pattern synonym be nice here? Could serve the purposes of both `termIn` and `termOut`.

Reply:
Maybe, but I don’t know what to call it, and I find pattern synonyms super janky and confusing and they interact strangely with imports and stuff, so I’d rather leave that for a future PR.

Also all of this is going to be burned to the ground ASAP so there’s that too.